tools/localization_analysis/scripts/analyze_gaps.py

```diff
@@ -0,0 +1,222 @@
+import rosbag
+import rospy
+import argparse
+import os
+import glob
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+import math
+import rospy
+import tf
+from geometry_msgs.msg import Quaternion
+
+matplotlib.use("pdf")
+from matplotlib.backends.backend_pdf import PdfPages
+
+def quat_to_rpy(quaternion):
```

Comment:
Do these functions exist in other files? If so, we should put them in a common place.